### ISSUE TITLE
No longer working on Youtube "home" page and subsections.

### ISSUE THREAD

**Issue by DNSDies** [sniklaus/youtube-watchmarker#133]
Youtube pushed some kind of update this morning which changed the visual style and now this no longer works in any of the "home" sections.
Tried a fresh install and brand new database and it's not greying out or marking videos as watched.
Still works in History, watch later, channel pages, and most other sections for now, just not in the main home page or the sub-categories and "new to you" section.

**Comment by VcSaJen** [comment]
Same here. Here's example of video thumb and title element:
```html
<ytd-rich-item-renderer class="style-scope ytd-rich-grid-renderer" items-per-row="4" lockup="true" rendered-from-rich-grid=""><!--css-build:shady--><!--css-build:shady--><div id="content" class="style-scope ytd-rich-item-renderer"><yt-lockup-view-model class="ytd-rich-item-renderer lockup"><div class="yt-lockup-view-model-wiz yt-lockup-view-model-wiz--vertical"><a href="/watch?v=ugH0YaBSaYk" class="yt-lockup-view-model-wiz__content-image" style="" aria-haspopup="false" tabindex="-1" aria-hidden="true"><yt-thumbnail-view-model class="yt-thumbnail-view-model yt-thumbnail-view-model--aspect-ratio-16-by-9 yt-thumbnail-view-model--large" style=""><div class="yt-thumbnail-view-model__image"><img alt="" class="yt-core-image yt-core-image--fill-parent-height yt-core-image--fill-parent-width yt-core-image--content-mode-scale-aspect-fill yt-core-image--loaded" src="https://i.ytimg.com/vi/ugH0YaBSaYk/hq720.jpg?sqp=-oaymwEcCOgCEMoBSFXyq4qpAw4IARUAAIhCGAFwAcABBg==&amp;rs=AOn4CLAlv45552NjQG2amRIHmzEp98L_2Q"></div><yt-thumbnail-overlay-badge-view-model class="yt-thumbnail-overlay-badge-view-model-wiz yt-thumbnail-overlay-badge-view-model-wiz--bottom-end yt-thumbnail-overlay-badge-view-model-wiz--large"><yt-thumbnail-badge-view-model class="yt-thumbnail-badge-view-model-wiz yt-thumbnail-overlay-badge-view-model-wiz__badge"><badge-shape class="badge-shape-wiz badge-shape-wiz--thumbnail-default badge-shape-wiz--thumbnail-badge"><div class="badge-shape-wiz__text">2:13</div></badge-shape></yt-thumbnail-badge-view-model></yt-thumbnail-overlay-badge-view-model></yt-thumbnail-view-model></a><div class="yt-lockup-view-model-wiz__metadata"><yt-lockup-metadata-view-model class="yt-lockup-metadata-view-model-wiz--vertical yt-lockup-metadata-view-model-wiz--standard yt-lockup-metadata-view-model-wiz"><div class="yt-lockup-metadata-view-model-wiz__avatar"><yt-decorated-avatar-view-model class="yt-decorated-avatar-view-model-wiz"><yt-avatar-shape><div class="yt-spec-avatar-shape yt-spec-avatar-shape__button yt-spec-avatar-shape__button--button-medium yt-spec-avatar-shape__button--tappable" aria-label="Go to channel" role="button" tabindex="0"><div class="yt-spec-avatar-shape--cairo-refresh"><div class="yt-spec-avatar-shape--avatar-size-medium"><img alt="" class="yt-core-image yt-spec-avatar-shape__image yt-core-image--fill-parent-height yt-core-image--fill-parent-width yt-core-image--content-mode-scale-to-fill yt-core-image--loaded" src="https://yt3.ggpht.com/B1VxysO6v-alAn_JDlVJi13rGLo8BpJBkwa5bTXhR4hqu9QSmSuE7Zd9-Im0KoIm6edmELAAMg=s68-c-k-c0x00ffffff-no-rj"><div class="yt-spec-avatar-shape__image-overlays yt-spec-avatar-shape__image"></div></div></div></div></yt-avatar-shape></yt-decorated-avatar-view-model></div><div class="yt-lockup-metadata-view-model-wiz__text-container"><h3 class="yt-lockup-metadata-view-model-wiz__heading-reset" title="COOL TIMES: BEAUTY"><a class="yt-lockup-metadata-view-model-wiz__title" style="" aria-label="COOL TIMES: BEAUTY by DeadlyComics 1,368,317 views 6 years ago 2 minutes, 13 seconds" href="/watch?v=ugH0YaBSaYk"><span class="yt-core-attributed-string yt-core-attributed-string--white-space-pre-wrap" role="text">COOL TIMES: BEAUTY</span></a></h3><div class="yt-lockup-metadata-view-model-wiz__metadata"><yt-content-metadata-view-model class="yt-content-metadata-view-model-wiz yt-content-metadata-view-model-wiz--medium-text"><div class="yt-content-metadata-view-model-wiz__metadata-row"><span class="yt-core-attributed-string yt-content-metadata-view-model-wiz__metadata-text yt-core-attributed-string--white-space-pre-wrap yt-core-attributed-string--link-inherit-color" dir="auto"><span class="" style="font-weight: 400;" dir="auto"><a class="yt-core-attributed-string__link yt-core-attributed-string__link--call-to-action-color yt-core-attributed-string--link-inherit-color" tabindex="0" href="/@DeadlyComicsYT" target="" force-new-state="true">DeadlyComics<span class="yt-core-attributed-string--link-inherit-color" style="color: rgb(96, 96, 96);" dir="auto"><span class="yt-core-attributed-string--inline-block-mod" style="margin-left: 4px;"><yt-icon style="height: 14px; width: 14px;" class="yt-core-attributed-string__image-element yt-core-attributed-string__image-element--image-alignment-vertical-center" aria-hidden="" aria-label="" role="img"><!--css-build:shady--><!--css-build:shady--><span class="yt-icon-shape style-scope yt-icon yt-spec-icon-shape"><div style="width: 100%; height: 100%; display: block; fill: currentcolor;"><svg xmlns="http://www.w3.org/2000/svg" height="24" viewBox="0 0 24 24" width="24" focusable="false" aria-hidden="true" style="pointer-events: none; display: inherit; width: 100%; height: 100%;"><path d="M12 2C6.5 2 2 6.5 2 12s4.5 10 10 10 10-4.5 10-10S17.5 2 12 2zM9.8 17.3l-4.2-4.1L7 11.8l2.8 2.7L17 7.4l1.4 1.4-8.6 8.5z"></path></svg></div></span></yt-icon></span></span></a></span></span></div><div class="yt-content-metadata-view-model-wiz__metadata-row"><span class="yt-core-attributed-string yt-content-metadata-view-model-wiz__metadata-text yt-core-attributed-string--white-space-pre-wrap yt-core-attributed-string--link-inherit-color" dir="auto" role="text">1.3M views</span><span class="yt-content-metadata-view-model-wiz__delimiter" aria-hidden="true"> • </span><span class="yt-core-attributed-string yt-content-metadata-view-model-wiz__metadata-text yt-core-attributed-string--white-space-pre-wrap yt-core-attributed-string--link-inherit-color" dir="auto" role="text">6 years ago</span></div></yt-content-metadata-view-model></div><div></div></div><div class="yt-lockup-metadata-view-model-wiz__menu-button"><button-view-model class="yt-spec-button-view-model"><button class="yt-spec-button-shape-next yt-spec-button-shape-next--text yt-spec-button-shape-next--mono yt-spec-button-shape-next--size-m yt-spec-button-shape-next--icon-button yt-spec-button-shape-next--enable-backdrop-filter-experiment" aria-disabled="false" style="" aria-label="More actions" title=""><div class="yt-spec-button-shape-next__icon" aria-hidden="true"><yt-icon style="width: 24px; height: 24px;"><!--css-build:shady--><!--css-build:shady--><span class="yt-icon-shape style-scope yt-icon yt-spec-icon-shape"><div style="width: 100%; height: 100%; display: block; fill: currentcolor;"><svg xmlns="http://www.w3.org/2000/svg" enable-background="new 0 0 24 24" height="24" viewBox="0 0 24 24" width="24" focusable="false" aria-hidden="true" style="pointer-events: none; display: inherit; width: 100%; height: 100%;"><path d="M12 16.5c.83 0 1.5.67 1.5 1.5s-.67 1.5-1.5 1.5-1.5-.67-1.5-1.5.67-1.5 1.5-1.5zM10.5 12c0 .83.67 1.5 1.5 1.5s1.5-.67 1.5-1.5-.67-1.5-1.5-1.5-1.5.67-1.5 1.5zm0-6c0 .83.67 1.5 1.5 1.5s1.5-.67 1.5-1.5-.67-1.5-1.5-1.5-1.5.67-1.5 1.5z"></path></svg></div></span></yt-icon></div><yt-touch-feedback-shape style="border-radius: inherit;"><div class="yt-spec-touch-feedback-shape yt-spec-touch-feedback-shape--touch-response" aria-hidden="true"><div class="yt-spec-touch-feedback-shape__stroke" style=""></div><div class="yt-spec-touch-feedback-shape__fill" style=""></div></div></yt-touch-feedback-shape></button></button-view-model></div></yt-lockup-metadata-view-model></div><yt-touch-feedback-shape style="border-radius: inherit;"><div aria-hidden="true" class="yt-spec-touch-feedback-shape yt-spec-touch-feedback-shape--touch-response yt-spec-touch-feedback-shape--is-extended yt-spec-touch-feedback-shape--trigger-events"><div class="yt-spec-touch-feedback-shape__stroke"></div><div class="yt-spec-touch-feedback-shape__fill"></div></div></yt-touch-feedback-shape></div></yt-lockup-view-model></div>
<yt-interaction id="interaction" class="extended rounded-large style-scope ytd-rich-item-renderer" hidden=""><!--css-build:shady--><!--css-build:shady--><div class="stroke style-scope yt-interaction"></div><div class="fill style-scope yt-interaction"></div></yt-interaction>
</ytd-rich-item-renderer>
```
XPath:
`/html/body/ytd-app/div[1]/ytd-page-manager/ytd-browse/ytd-two-column-browse-results-renderer/div[1]/ytd-rich-grid-renderer/div[6]/ytd-rich-item-renderer[2]`
CSS selector:
```
html body ytd-app div#content.style-scope.ytd-app ytd-page-manager#page-manager.style-scope.ytd-app ytd-browse.style-scope.ytd-page-manager ytd-two-column-browse-results-renderer.style-scope.ytd-browse.grid.grid-disabled div#primary.style-scope.ytd-two-column-browse-results-renderer ytd-rich-grid-renderer.style-scope.ytd-two-column-browse-results-renderer div#contents.style-scope.ytd-rich-grid-renderer ytd-rich-item-renderer.style-scope.ytd-rich-grid-renderer
```

**Comment by XP1** [comment]
@VcSaJen, I just got the new update for the related videos sidebar. Here on my branch, I only needed to add one line:

```js
"a.yt-lockup-view-model-wiz__content-image[href^=\"/watch?v=" + id + "\"]",
```

Here is the commit:
Fix related videos sidebar. https://github.com/XP1/youtube-watchmarker/commit/b92a4f57d9ba86c20f2edb9041f7b055a4a34557

It's commited into my branch [feature/mark-video-player-and-shorts](https://github.com/XP1/youtube-watchmarker/tree/feature/mark-video-player-and-shorts) and folded into my PR https://github.com/sniklaus/youtube-watchmarker/pull/126.

**Comment by sniklaus** [comment]
Thanks to the contributions from @XP1, I just released version `4.7.0` (already live on Firefox, still pending on Chrome) which should fix this. I am hence closing this case for now, but please don't hesitate to reopen it should the issue persist. :+1:
